src/components/forms/checkbox/checkbox.tsx

```diff
@@ -70,29 +67,19 @@ export const Checkbox = (
 ) => {
   const { field, fieldState, formState } = useController(props);
   const errorMessage = fieldState?.error?.message;
-  const [checked, setChecked] = useState(field?.value || false);
-
-  const handleChange = () => {
-    const newChecked = !checked;
-    setChecked(newChecked);
-    field.onChange(newChecked);
-  };
 
   return (
-    <div>
+    <div style={{ position: 'relative' }}>
       <CheckboxLabel>
-        <HiddenCheckbox
+        <StyledCheckbox
           aria-required={true}
           disabled={formState.isSubmitting}
           {...field}
-          value={field?.value || false}
           {...props}
           id={props.name}
-          onChange={handleChange}
+          hasError={Boolean(fieldState?.error)}
```

Comment:
Can you please change the name to `$hasError`? The `$`-prefix tells styled-components to not pass the prop to the DOM Element ([reference here](https://styled-components.com/docs/basics#adapting-based-on-props)). Otherwise, React isn't happy 😕
![Bildschirmfoto 2024-09-13 um 11 10 48](https://github.com/user-attachments/assets/2e36cec7-f9f0-4aa8-9ad9-87d288039dc5)